### ISSUE TITLE
Option to suppress error for type hint comments on `with` and `for` statements

### ISSUE THREAD

**Issue by OldSneerJaw** [microsoft/pyright#3602]
**Is your feature request related to a problem? Please describe.**
A recent [change](https://github.com/microsoft/pyright/commit/9b407f6c9e2a237491c4f5a8dbbe5718aaf757bf) made it an error to use a `# type: xxx` comment on `with` and `for` statements. As I understand it, this is consistent with PEP 484. However, mypy allows and respects `# type: xxx` on those statements and, for a project that relies on mypy exclusively for type checking and includes imports from external libraries with nonexistent/incomplete type annotations, I think it is cruel for VS Code/Pylance/Pyright to report an error that is not considered such by the other tools and with no way to suppress it.

**Describe the solution you'd like**
A type check diagnostic config option to set the diagnostic level for reporting `annotationNotSupported` diagnostic errors. Suggested name: `reportAnnotationNotSupported`.

**Additional context**
As an example, consider a hypothetical external library with a function called `connect` whose return value is a context manager. Now consider the case where `connect` has no return type annotation (i.e. return value is inferred to be `Any`). Here's what the code might look like:

```python
with connect(connection_string) as db_connection:  # type: DbConnection
    return db_connection.execute_query(sql_query, sql_params)
```

While some might argue against using this style, we have already established that mypy considers the style valid and it enforces the type constraints when the style is used. Therefore, I believe in such cases it should be possible to override Pyright's opinion through configuration.

**Comment by erictraut** [comment]
You mentioned that you rely exclusively on mypy for type checking. Does that mean that you are not using pyright for type checking? In other words, you have all type checking diagnostic rules disabled including `reportGeneralTypeIssues`? I think it's reasonable to move the `annotationNotSupported` diagnostic errors under that diagnostic rule so it is disabled if you are not using pyright or pylance for type checking. Does that meet your needs?

**Comment by erictraut** [comment]
I've moved the `annotationNotSupported` diagnostics under the `reportGeneralTypeIssues` diagnostic rule. If you are not using pyright for type checking (i.e. you have all of the diagnostic rules disabled or have "typeCheckingMode" set to "off"), then you won't see these diagnostics.

**Comment by OldSneerJaw** [comment]
Those are all good points, and I don’t necessarily disagree with them. As for moving `annotationNotSupported` into the `reportGeneralTypeIssues` diagnostic rule, I think that makes sense. In fact, that was one of the first things I tried, to no avail.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.256, which I just published. It will also be included in the next release of pylance.
